app/main.py

```diff
@@ -19,7 +19,7 @@ def __repr__(self) -> str:
 
 class Herbivore(Animal):
     def hide(self) -> None:
-        if self.hidden is False:
+        if not self.hidden:
             self.hidden = True
         else:
             self.hidden = False
```

Comment:
I meant the whole structure from 22 to 25 line. you can do it without if statements